### PR TITLE
[extractor/sovietscloset] Fix __NUXT_JSONP__ extraction

### PR DESCRIPTION
**Pull request by ChillingPepper**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

</details>

This partially reverts https://github.com/yt-dlp/yt-dlp/commit/f55523cfdd18dcd578f5d96cbb06266663169d35 which JSON encoded all `js_to_json` vars. This broke \_\_NUXT_JSONP\_\_ parsing because JS numbers turned into strings (`123` to `"123"`) and JS strings turned into quoted strings (`"abc"` to `"\"abc\""`).

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [x] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

File: test/test_utils.py [hunk 1]
```diff
@@ -954,6 +954,88 @@ def test_escape_url(self):
...
+                    'falseStr': f,
+                }''',
+                {
+                    'a': 'null',
+                    'b': '"null"',
+                    'c': 'true',
+                    'd': '"true"',
+                    'e': 'false',
+                    'f': '"false"',
+                }
```
Comment (pukkandan) [inline test/test_utils.py:981]: The tests are wrong? This should be
```suggestion
                {
                    'a': null,
                    'b': 'null',
                    'c': True,
                    'd': 'true',
                    'e': False,
                    'f': 'false',
                }
```

**Comment by pukkandan** [comment]
Please post verbose log of the error you are getting. I suspect it's an extractor bug

File: test/test_utils.py (hunk 1 above)
Comment (ChillingPepper) [reply test/test_utils.py:981]: These are the variables that are passed to `js_to_json` as a dict. The values are what would have been extracted from a JS string, so `a` and `b` could come from `someFunction(null, "null")`.

**Comment by ChillingPepper** [comment]
> I see the issue, but we need to think about how to deal with #4900 when fixing this

The Skeb extractor seems to be broken at the moment so I don't know how to fix it.

**Comment by ChillingPepper** [comment]
Skeb appears to have a new structure, using a JSON API instead of \_\_NUXT_JSONP\_\_ (for example `https://skeb.jp/api/users/riiru_wm/works/10` for `https://skeb.jp/@riiru_wm/works/10`).

**Comment by pukkandan** [comment]
Let's leave skeb itself for another PR. I have modified your test to simulate the underlying issue which https://github.com/yt-dlp/yt-dlp/commit/f55523cfdd18dcd578f5d96cbb06266663169d35 was supposed to solve

File: test/test_utils.py [hunk 2]
```diff
@@ -954,6 +954,91 @@ def test_escape_url(self):
         )
         self.assertEqual(escape_url('http://vimeo.com/56015672#at=0'), 'http://vimeo.com/56015672#at=0')
 
+    def test_js_to_json_vars_strings(self):
+        def parse_js_to_json(js, vars):
+            try:
+                return json.loads(js_to_json(js, vars))
+            except json.JSONDecodeError as e:
+                raise ValueError(f'JSON: {e.msg}: line {e.lineno} column {e.colno} "{e.doc}"')
```
Comment (pukkandan) [inline test/test_utils.py:962]: why the throw and catch?

Comment (ChillingPepper) [reply test/test_utils.py:962]: For line and column numbers with the JSON document, it's not necessary, just nice to have if the test breaks.

**Comment by ChillingPepper** [comment]
> I have modified your test to simulate the underlying issue which [f55523c](https://github.com/yt-dlp/yt-dlp/commit/f55523cfdd18dcd578f5d96cbb06266663169d35) was supposed to solve

How is it supposed to work? If the JS contains an unresolved variable `js_to_json` can't turn it into valid JSON. Is the goal to make it parseable no matter what?

**Comment by ChillingPepper** [comment]
Should this be behind a flag? Maybe `js_to_json(..., lenient=False)`?

**Comment by ChillingPepper** [comment]
> Should this be behind a flag? Maybe `js_to_json(..., lenient=False)`?

It's now behind the existing `strict` flag. I think this is done @pukkandan.

**Pull request merged**